### PR TITLE
search movies made

### PR DESCRIPTION
**Pull request by briansetzesglobant**
Perform search implementation for a particular movie.
And assign a default image in case you don't have a poster.

https://user-images.githubusercontent.com/84338449/122570788-7710fe80-d022-11eb-880b-b31ff1abc77a.mp4

### CODE REVIEW

File: lib/src/models/movie_result.dart [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import 'package:movies/utils/constants.dart';
```
Comment (FranciscoGaraventa) [inline lib/src/models/movie_result.dart]: Use relative import here

File: lib/src/models/movie_result.dart [hunk 2]
```diff
@@ -14,14 +16,16 @@ class MovieResult {
...
     _vote_count = result['vote_count'];
     _id = result['id'];
     _video = result['video'];
     _vote_average = result['vote_average'];
     _title = result['title'];
     _popularity = result['popularity'];
-    _poster_path = result['poster_path'];
+    _poster_path = result['poster_path'] != null
+        ? Constants.imageNetwork + result['poster_path']
+        : "https://kubalubra.is/wp-content/uploads/2017/11/default-thumbnail.jpg";
```
Comment (FranciscoGaraventa) [inline lib/src/models/movie_result.dart]: This url can be added to Constants file

File: lib/src/resources/movies_repository.dart [hunk 3]
```diff
@@ -5,5 +5,10 @@ import '../models/movie_model.dart';
 class MoviesRepository {
```
Comment (FranciscoGaraventa) [inline lib/src/resources/movies_repository.dart]: This repository should also be implementing an interface

File: lib/src/ui/home_page.dart [hunk 4]
```diff
@@ -36,9 +37,57 @@ class HomePageState extends State<HomePage> {
...
+                          color: Colors.black,
+                        ),
+                      ),
+                    ),
+                    onSubmitted: (nameMovie) {
+                      widget.bloc.searchByMovieName(nameMovie);
+                    },
+                  ),
+                  width: Constants.widthSearch,
+                  height: Constants.heightSearch,
```
Comment (FranciscoGaraventa) [inline lib/src/ui/home_page.dart]: You can change this with a more descriptive constant name like searchSixedBoxWidth and searchSixedBoxHeight

File: lib/utils/constants.dart [hunk 5]
```diff
@@ -1,11 +1,22 @@
 abstract class Constants {
```
Comment (FranciscoGaraventa) [inline lib/utils/constants.dart:1]: As a suggestion you can create different files, one relative to all constants that you have in code, another one that only have Strings or something relative to repository/api provider

**Review by MarcosCGlobant — COMMENTED** [review]
Approve after Fran's comments and also check Javi's comments on Horacio's PR, so you all can have a similar convention for the projects https://github.com/HoracioDuca/movies_bloc/pull/1

File: lib/src/models/movie_result.dart (hunk 1 above)
Comment (briansetzesglobant) [reply lib/src/models/movie_result.dart]: done

File: lib/src/models/movie_result.dart (hunk 2 above)
Comment (briansetzesglobant) [reply lib/src/models/movie_result.dart]: done

File: lib/src/resources/movies_repository.dart (hunk 3 above)
Comment (briansetzesglobant) [reply lib/src/resources/movies_repository.dart]: fixed

File: lib/src/ui/home_page.dart (hunk 4 above)
Comment (briansetzesglobant) [reply lib/src/ui/home_page.dart]: fixed

File: lib/utils/constants.dart (hunk 5 above)
Comment (briansetzesglobant) [reply lib/utils/constants.dart:1]: done

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

File: lib/src/resources/movie_repository.dart [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import 'dart:async';
+import 'package:movies/src/resources/i_movie_repository.dart';
```
Comment (MarcosCGlobant) [inline lib/src/resources/movie_repository.dart]: relative import here, please

File: lib/src/resources/movie_api_provider.dart [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  Future<MovieModel> searchByMovieName(String nameMovie) async {
+    final _response = await client.get(
+      Uri.parse(
+        MovieStrings.uriSearchClient + nameMovie,
+      ),
+    );
+    return resp(_response);
+  }
+
+  MovieModel resp(http.Response _response) {
```
Comment (MarcosCGlobant) [inline lib/src/resources/movie_api_provider.dart]: "resp" is not a descriptive name, avoid cropping words as much as posible when naming elements

File: lib/src/resources/i_movie_repository.dart [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import '../models/movie.dart';
+
+abstract class IMovieRepository {
+  Future<MovieModel> fetchAllMovies();
```
Comment (MarcosCGlobant) [inline lib/src/resources/i_movie_repository.dart]: you change the name of the file movie_model to movie, but the class hasn't change, rename it to keep consistency

File: lib/src/resources/movie_repository.dart (hunk 6 above)
Comment (briansetzesglobant) [reply lib/src/resources/movie_repository.dart]: done

File: lib/src/resources/movie_api_provider.dart (hunk 7 above)
Comment (briansetzesglobant) [reply lib/src/resources/movie_api_provider.dart]: done

File: lib/src/resources/i_movie_repository.dart (hunk 8 above)
Comment (briansetzesglobant) [reply lib/src/resources/i_movie_repository.dart]: done

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by marianosalvetti — APPROVED** [review]

**Pull request merged**